test/hex_web/block_address_test.exs

```diff
@@ -0,0 +1,52 @@
+defmodule HexWeb.BlockAddressTest do
+  use ExUnit.Case, async: false
+  use Plug.Test
+
+  defmodule Hello do
+    use Phoenix.Controller
+
+    plug :accepts, ~w(json)
+    plug HexWeb.BlockAddress.Plug
+
+    def index(conn, _params) do
+      send_resp(conn, 200, "Hello, World!")
+    end
+  end
+
+  test "allows requests from IPs that are not blocked" do
+    conn = request({0, 0, 0, 0})
+    assert conn.status == 200
+    assert conn.resp_body == "Hello, World!"
+  end
+
+  test "halts requests from IPs that are blocked" do
+    HexWeb.Repo.insert!(%HexWeb.BlockedAddress{ip: "1.1.1.1", comment: "blocked"})
+    HexWeb.BlockAddress.reload
```

Comment:
one thing that I couldn't find out in code is where `BlockAddress` is told to re-fetch data from the DB (by setting `loaded: false`) so that's why I'm manually reloading it in tests